src/ansys/additive/core/simulation_task.py

```diff
@@ -0,0 +1,301 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Container for a simulation task."""
+
+from __future__ import annotations
+
+import os
+import zipfile
+
+from ansys.api.additive.v0.additive_materials_pb2 import TuneMaterialResponse
+from ansys.api.additive.v0.additive_operations_pb2 import OperationMetadata
+from ansys.api.additive.v0.additive_simulation_pb2 import SimulationResponse
+from google.longrunning.operations_pb2 import GetOperationRequest, Operation, WaitOperationRequest
+from google.protobuf.any_pb2 import Any
+from google.protobuf.duration_pb2 import Duration
+from google.rpc.status_pb2 import Status as RpcStatus
+
+from ansys.additive.core.download import download_file
+from ansys.additive.core.material_tuning import MaterialTuningInput, MaterialTuningSummary
+from ansys.additive.core.microstructure import MicrostructureInput, MicrostructureSummary
+from ansys.additive.core.microstructure_3d import Microstructure3DInput, Microstructure3DSummary
+from ansys.additive.core.porosity import PorosityInput, PorositySummary
+from ansys.additive.core.progress_handler import IProgressHandler, Progress, ProgressState
+from ansys.additive.core.server_connection import ServerConnection
+from ansys.additive.core.simulation import SimulationError
+from ansys.additive.core.single_bead import SingleBeadInput, SingleBeadSummary
+from ansys.additive.core.thermal_history import ThermalHistoryInput, ThermalHistorySummary
+
+
+class SimulationTask:
+    """Provides a simulation task.
+
+    Parameters
+    ----------
+    server_connection: ServerConnection
+        The client connection to the Additive server.
+    long_running_operation: Operation
+        The long running operation representing the simulation on the server.
+    simulation_input: SingleBeadInput | PorosityInput | MicrostructureInput | ThermalHistoryInput | Microstructure3DInput | MaterialTuningInput
+        The simulation input.
+    user_data_path: str
+        The path to the user data directory.
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        server_connection: ServerConnection,
+        long_running_operation: Operation,
+        simulation_input: (
+            SingleBeadInput
+            | PorosityInput
+            | MicrostructureInput
+            | ThermalHistoryInput
+            | Microstructure3DInput
+            | MaterialTuningInput
+        ),
+        user_data_path: str,
+    ):
+        """Initialize the simulation task."""
```

Comment:
```suggestion
        """Start the simulation."""
```

Reply:
@julieatansys  This doesn't actually start a simulation, this is the setup for a class that stores the simulation input, the server to run on, and the operation handle that has been returned by the server.

Reply:
> @julieatansys This doesn't actually start a simulation, this is the setup for a class that stores the simulation input, the server to run on, and the operation handle that has been returned by the server.

Yeah, I figured it might be after reading all the task stuff later in the PR.